library/core/class.request.php

```diff
@@ -897,6 +897,30 @@ protected function _setRequestArguments($paramsType, $paramsData = null) {
         $this->_RequestArguments[$paramsType] = $argumentData;
     }
 
+    /**
+     * Decode the environment's post depending on content type.
+     *
+     * @param array $post Usually the {@link $_POST} super-global.
+     * @param array $server Usually the {@link $_SERVER} super-global.
+     * @param array $inputFile Usually **php://input** for the raw input stream.
+     */
+    private function decodePost($post, $server, $inputFile) {
```

Comment:
My thinking here is `$inputFile` is almost never going to be anything other than "php://input". It's great that we can change that, for things like automated testing, but otherwise it's probably always going to be PHP's raw-body stream.

With this function being private and only called in one place, it doesn't matter much.  It just struck me as a good candidate for a default value.